### DISCUSSION TITLE
No connection over WiFi after installing Google Nest WiFi Pro stations

### DISCUSSION THREAD

**Discussion by HaViGit** [arendst/Tasmota#21441] [Support request] [answered]
In my home network I use several Sonoff switches that I have flashed with the latest version of Tasmota. Everything worked well until I started using Google Nest WiFi Pro with an update to my network. I can no longer connect to the Sonoff switches.

Are there any known problems with Tasmota in combination with Nest WiFi Pro Mesh stations? Does anyone have a solution?

**Comment by sfromis** [comment]
I've had no issues connecting to Nest Wifi pro, of course having ensured correct and supported Wifi credentials.

**Reply by HaViGit** [reply]
Of course the WiFi settings are also correct here ;-)

Have you not set anything special via the Home app?

**Reply by sfromis** [reply]
No special things done, I just selected the different AP from inside Tasmota.

**Comment by HaViGit** [comment]
Below the output from Termite terminal, regardless of whether it is powered with 230 volts or via FTDI interface:

00:01:53.785 WIF: Connect to AP1 TIS in mode 11n as tasmota-740BE7-3047...
00:02:08.773 WIF: Connection failed because AP is unreachable

**Reply by sfromis** [reply]
Maybe try if it helps doing a `Reset 3` command, followed by power cycling.
This erases Wifi calibration data, but leaves settings untouched.

**Comment by HaViGit** [comment]
Unfortunately a "Reset 3" didn't help, just to be sure I also re-flashed Tasmota to one of the Sonoff switches but that didn't help either :-(

**Comment by HaViGit** [comment]
_sfromis wrote: No special things done, I just selected the different AP from inside Tasmota._

Maybe I'm making a mistake, in the terminal I enter 192.168.2.254 for the Tasmota/Sonoff Switch as Gateway IP (IPAddress2), that is my modem-router address.

But maybe I should provide an IP address of the Nest WiFi Pro there? And if so, how do I find that IP address and is that address static?

**Reply by sfromis** [reply, marked as answer]
What I'm doing is not inputting *any* IP addresses in Tasmota, but let it all be handled via DHCP. If I want a "stable" IP address, I reserve such on the router DHCP. Anyway, setting the gateway IP does not "help" you (more likely harming you), unless you actively want to avoid using DHCP.

When using DHCP, I also have the convenience factor that the Tasmota host name also gets found in DNS lookup, as the router is using dnsmasq to provide DHCP+DNS. Hence I need not really worry about the IP address, even if I do tend to reserve it in DHCP.

**Comment by HaViGit** [comment]
First of all, I made a mistake about my network settings. I was used to using a combined modem/router (Fritz!Box 7590), but now that I have a Nest WiFi Pro set, the extra router is used for wireless traffic.

I would prefer to use the Sonoff switches such as sfronis, so everything handled via DHCP. Unfortunately I can't manage that. These are the steps:

1) With Windows Tasmotizer I flash the NL version of Tasmota
2) After a power cycle I can connect to the Sonoff switch using the Termite app
3) With "SSID1 ssidname" I specify the SSID of the Nest WiFi Pro
4) With "Password1 mypass" I enter the WiFi password of the Nest WiFi Pro

The switch generates an error message, the connection fails because the AP is not reachable:

<img width="448" alt="1" src="https://github.com/arendst/Tasmota/assets/46343309/9ca56f83-4532-43a2-8beb-3d9b69f3ab35">

Then I try "IPAdress1 0.0.0.0" to force DHCP (sfronis) but that gives the same result. I really don't know what to do now...

For completeness, the Nest WiFi Pro's Subnet Mask is 255.255.255.0 and the Default Gateway is 192.168.86.1. A screenshot of the LAN settings:

![2](https://github.com/arendst/Tasmota/assets/46343309/5ee71218-9c11-4ee6-be0e-a09a2a617b84)

**Reply by sfromis** [reply]
If you want a setup easy to manage, I'd suggest having the Nest Wifi Pro router handle all the routing and network, with the Fritzbox merely bridging Internet traffic to/from the "main router".

If you want to have two routers doing "their" full functions at the same time, you need to set up routing of IP traffic, as a device on the "outer" network (Fritzbox) cannot "see" a device managed by the Nest Wifi Pro, when the DHCP+NAT is done there, unless you can set up routing tables.

I haven't experimented with what can (or not) be done with the settings on the Nest Wifi Pro, like if it is feasible to deactivate DHCP/NAT, to basically leave it to only act as an AP mesh, but Google has clearly opted for excluding the complications of detailed configuration features, to instead have it be easy to set up and low risk of getting in trouble - as long as you use it as designed.

**Comment by HaViGit** [comment]
Thanks again for your response! Maybe a bit off-topic for Tasmota, but I'm curious anyway. You wrote that Nest Pro should do all the routing, does this mean that I can set that in the internet modem/router? Or can the Nest Pro only route non-wired traffic?

PS:
I tried flashing a Sonoff Smart Socket and using it on the Nest Pro and the first impression is that it works well...

**Reply by sfromis** [reply]
What you can do depends on your setup around that Fritzbox, if this is required by your ISP, you should be able to connect the Nest Wifi Pro to an Ethernet port, and normal usage should work right away, with the Fritzbox only seeing one device on the network, when it is providing a WAN connection for the Nest Wifi Pro. If you connect more to the Fritzbox, this cannot see any IP managed by the Nest Wifi Pro, unless you are able to set up routing tables for such.

What would be "lost" with this basic setup would be if you want to serve incoming connections, like a web or email server. Etc. Also UDP from the outside, I suppose. Such would have to be port forwarded to the Nest Wifi Pro, or maybe have it configured as a DMZ, routing all incoming traffic to the primary router.

Of course, if you are not accepting incoming connections, things are pretty simple.

**Comment by HaViGit** [comment]
Thanks again for your information, as an inexperienced user of the Nest WiFi Pro you helped me a lot.

The problems with connecting the Sonoff/Tasmota switches seem to have been solved, the Sonoff Basic with which I initially tested everything seems to be broken because other switches work well straight away. I should have tried another one straight away...

I also followed your advice to have the switches obtain an IP address via DHCP. In the Nest WiFi Pro (Home) app it was indeed very easy to reserve an IP address based on the MAC address.

**Reply by sfromis** [reply]
It is common that old Sonoff Basic devices are broken, barely able to power themselves when running Tasmota.
